tedana/viz.py

```diff
@@ -46,41 +76,47 @@ def write_comp_figs(ts, mask, comptable, mmix, n_vols,
         Reference image to dictate how outputs are saved to disk
 
     """
-
+    # Check that colormap provided exists
+    if png_cmap not in plt.colormaps():
+        LGR.info('Provided colormap is not valid, proceeding with coolwarm')
+        png_cmap = 'coolwarm'
     # regenerate the beta images
     ts_B = model.get_coeffs(ts, mmix, mask)
     ts_B = ts_B.reshape(ref_img.shape[:3] + ts_B.shape[1:])
-    # Mask out zeros
+    # trim edges from ts_B array
+    ts_B = trim_edge_zeros(ts_B)
+
+    # Mask out remaining zeros
     ts_B = np.ma.masked_where(ts_B == 0, ts_B)
 
     # Get repetition time from ref_img
     tr = ref_img.header.get_zooms()[-1]
 
-    # Start making plots
-    if not os.path.isdir('figures'):
-        os.mkdir('figures')
-
     # Create indices for 6 cuts, based on dimensions
     cuts = [ts_B.shape[dim] // 6 for dim in range(3)]
-
+    expl_text = ''
     for compnum in range(0, mmix.shape[1], 1):
 
         if compnum in acc:
             line_color = 'g'
+            expl_text = 'accepted'
         elif compnum in rej:
             line_color = 'r'
+            expl_text = 'rejection reason: ' + comptable.iloc[compnum]["rationale"]
```

Comment:
reason(s) or rationale(s). There can be more than one reason for rejection.

Reply:
You'll also need to strip off trailing semicolons.

Reply:
edited, but have not changed the semicolons, not quite sure how to do that (without removing the middle semicolons).

Reply:
It turns out `rstrip` accepts characters, so you can do `comptable.iloc[compnum]["rationale"].rstrip(';')`. Alternatively, it might be worth it to do that in selcomps so that we don't have to deal with this separately in each location where we reference rationales. Something like `comptable['rationale'] = comptable['rationale'].str.rstrip(';')` at the end of selcomps would probably work.

Reply:
Trying rstrip now.

Reply:
I remove the trailing ';'  with str.rstrip prior to the loops, so it is all done at once, rather than over and over. Seems to work well, and doesn't interfere with multiple rationales.